engines/suppliers/app/controllers/suppliers/application_controller.rb

```diff
@@ -1,4 +1,16 @@
 module Suppliers
   class ApplicationController < ActionController::Base
+    include ::Shared::Controller::Authentication
+
+    rescue_from 'ActiveRecord::InvalidForeignKey' do |exception|
+      adapter = ExceptionAdapter.new
+      render json: ::Shared::Error.new(adapter, name: :bad_request)
+    end
+
+    class ExceptionAdapter
+      def errors
+        'The provided id does not exist'.freeze
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.